malloc/malloc.c

```diff
@@ -43,7 +60,36 @@ void *mem_copy(void *dest, const void *src, int n)
 */
 void *resize_memory(void *ptr, int old_size, int new_size)
 {
+    if (new_size == 0)
+    {
+        free(ptr);
+        return NULL;
+    }
+
+    else if (!ptr)
+    {
+        return malloc(new_size);
+    }
+
+    else if (old_size == new_size)
+    {
+        return ptr;
+    }
 
+    void *new_block = malloc(new_size);
+
```

Comment:
can you think of anything allocating memory for would be useful in a real world application?